### PR TITLE
[Documentation] Search bar for tables within docs

### PR DESCRIPTION
**Pull request by fraser-combe**
🗑️ This dev branch should be deleted after merging to main.

## :brain: Summary
This PR adds a search box to the tables within the documentation to improve the user experience when navigating through input and output details.

example markdown table updates wrapping in div with specific class

````
<div class="searchable-table" markdown="1">
Name | Value
-- | --
Item 1 | 100
Item 2 | 200
Item 3 | 300

</div>
````

<div class="searchable-table" markdown="1">: The div element wraps around the table. The class="searchable-table" is used to apply custom styles or JavaScript to this specific container. The attribute markdown="1" tells the Markdown processor to parse the content inside the div as Markdown.

## :zap: Impacted Workflows/Tasks

Documentation: Added search functionality for the input and output tables.

This PR may lead to different results in pre-existing outputs: No

This PR uses an element that could cause duplicate runs to have different results: No

## :hammer_and_wrench: Changes
Added a search box above the tables for input and output data to allow filtering table content.
Updated the CSS to ensure the search box styles are applied 
Modified the JavaScript to dynamically add search boxes to relevant tables.

### :gear: Algorithm
N/A. The changes are related to the documentation display and user interface improvements.

### ➡️ Inputs
N/A

### ⬅️ Outputs
N/A

## :test_tube: Testing
Verified that the search box appears above the input and output tables.
Tested the search functionality to ensure filtering works correctly
Tested search box is present upon page refresh

### Suggested Scenarios for Reviewer to Test
Verify that the search box appears above both input and output tables.
Test the search functionality by entering various keywords and ensuring the filtering behaves as expected.

## :microscope: Final Developer Checklist

- [ ] The workflow/task has been tested and results, including file contents, are as anticipated
- [ ] The CI/CD has been adjusted and tests are passing (Theiagen developers)
- [X] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [ ] Documentation and/or workflow diagrams have been updated if applicable (Theiagen developers only)

## 🎯 Reviewer Checklist

- [ ] All changed results have been confirmed
- [ ] You have tested the PR appropriately (see the [testing guide](https://theiagen.notion.site/PR-Testing-Guide-Determining-Appropriate-Levels-of-Testing-4764e98a6aeb460185039c0896714590) for more information)
- [ ] All code adheres to the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [ ] MD5 sums have been updated
- [ ] The PR author has addressed all comments
- [ ] The documentation has been updated

### CODE REVIEW

**Comment by sage-wright** [comment]
please remove all items in the site/ folder

File: mkdocs.yml [hunk 1]
```diff
@@ -230,11 +230,16 @@ plugins:
   # - section-index
 
 extra_javascript:
-  - https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js
-  - javascripts/tablesort.js
+  - https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js
+  - https://cdn.datatables.net/1.13.4/js/jquery.dataTables.min.js
```
Comment (sage-wright) [inline mkdocs.yml:234]: do you still need the datatables jquery scripts?

File: mkdocs.yml [hunk 2]
```diff
@@ -230,11 +230,16 @@ plugins:
...
+  - https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js
+  - https://cdn.datatables.net/1.13.4/js/jquery.dataTables.min.js  
+  - https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js       
+  - javascripts/tablesort.js                                      
+  - assets/js/table-search.js                                     
 
 extra_css:
-  - stylesheets/extra.css
+  - stylesheets/extra.css                                         
+  - https://cdn.datatables.net/1.13.4/css/jquery.dataTables.min.css
```
Comment (sage-wright) [inline mkdocs.yml:241]: same for this one

File: docs/assets/js/table-search.js
Comment (sage-wright) [inline docs/assets/js/table-search.js:1]: would a better location for this file be in the docs/javascripts folder?

File: docs/stylesheets/extra.css [hunk 3]
```diff
@@ -185,4 +185,18 @@ td {
   word-break: break-all;
 }
 
+/* Style the search box for tables*/
+input[type="search"] {
+  width: 25%;            
+  padding: 8px;           
+  margin-bottom: 15px;    
+  font-size: 16px;        
+  border: 1px solid #ccc;
```
Comment (sage-wright) [inline docs/stylesheets/extra.css:194]: could you insert the hex code I have you for this border color?

File: mkdocs.yml (hunk 1 above)
Comment (sage-wright) [reply mkdocs.yml:234]: I think you deleted the tablesort javascripts here on accident, which removes the ability to sort columns.  

Could you re-add
```  
  - https://unpkg.com/tablesort@5.3.0/dist/tablesort.min.js
  - javascripts/tablesort.js
```

then, could you check if you need these two lines:

```
  - https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js
  - https://cdn.datatables.net/1.13.4/js/jquery.dataTables.min.js  
```

My guess is that you may not need the `- https://cdn.datatables.net/1.13.4/js/jquery.dataTables.min.js`

Thanks!

**Comment by fraser-combe** [comment]
updated extra_ javascript for table sort functionality

File: docs/stylesheets/extra.css [hunk 4]
```diff
@@ -184,5 +184,36 @@ th {
...
+  font-size: 12px;
+  box-sizing: border-box;
+  border-radius: 2px; 
+}
+
+/* Light mode styles */
+[data-md-color-scheme="light"] div.searchable-table input.table-search-input {
+  background-color: #fff;
+  color: #000;
+  border: 1px solid #ccc;
```
Comment (sage-wright) [inline docs/stylesheets/extra.css:201]: can you change this to be 
`border: 1px solid #E0E1E1;`

**Review by sage-wright — APPROVED** [review]
This will be so usefull!!! 🌟

**Pull request merged**